catalog/resource_grants.go

```diff
@@ -316,6 +328,9 @@ func ResourceGrants() common.Resource {
 			}
 			var grants PermissionsList
 			common.DiffToStructPointer(d, s, &grants)
+			if firstPassOfForcedNew(d) {
+				return nil
+			}
 			return mapping.validate(d, grants)
```

Comment:
After looking at this code for 2.5 hours... I think I finally realized what the core issue is and that the actual bug is in `securableMapping.validate()`, specifically `kv()`. The issue is that the new value of the securable may not be known at plan time, but we still should know the type of the securable. That is sufficient for validation. This means that in `kv()`, instead of getting the value of the field, we can check the raw config to see which field is not nil.
![grant_validation](https://github.com/databricks/terraform-provider-databricks/assets/1850319/c71f6ca2-2d2b-4f7d-b6dc-304e3364be7e)